pkg/chaosctl/cmd/completion.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+// completionCmd represents the completion command
+var completionCmd = &cobra.Command{
+	Use:   "completion [bash|zsh|fish|powershell]",
+	Short: "Generate completion script",
+	Long: `To load completions:
+
+Bash:
+
+$ source <(chaosctl completion bash)
+
+# To load completions for each session, execute once:
+Linux:
+  $ chaosctl completion bash > /etc/bash_completion.d/chaosctl
+MacOS:
+  $ chaosctl completion bash > /usr/local/etc/bash_completion.d/chaosctl
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ chaosctl completion zsh > "${fpath[1]}/_chaosctl"
+
+# You will need to start a new shell for this setup to take effect.
+
+Fish:
+
+$ chaosctl completion fish | source
+
+# To load completions for each session, execute once:
+$ chaosctl completion fish > ~/.config/fish/completions/chaosctl.fish
+`,
+	DisableFlagsInUseLine: true,
+	ValidArgs:             []string{"bash", "zsh", "fish", "powershell"},
+	Args:                  cobra.ExactValidArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		switch args[0] {
+		case "bash":
+			cmd.Root().GenBashCompletion(os.Stdout)
+		case "zsh":
+			cmd.Root().GenZshCompletion(os.Stdout)
+		case "fish":
+			cmd.Root().GenFishCompletion(os.Stdout, true)
+		case "powershell":
+			cmd.Root().GenPowerShellCompletion(os.Stdout)
+		}
```

Comment:
So have you tested `fish` and `powershell` support in Cobra? If they are well supported already then it is fine.

Reply:
I just test for `fish` which works fine. 
Since I don't have k8s on windows, I test `gh`(github cli) which use [the current version of cobra](https://github.com/cli/cli/blob/4a8ab49cf1d23c76d77a7cb69537b2bc66769afa/pkg/cmd/completion/completion.go) and it works fine. Good to go~